docs/tutorials/tabular_prediction/tabular-indepth.md

```diff
@@ -46,159 +57,277 @@ gbm_options = { # specifies non-default hyperparameter values for lightGBM gradi
+for i in range(num_test):
+    datapoint = test_data_nolabel.iloc[[i]]
+    pred_numpy = predictor.predict(datapoint)
+    preds[i] = pred_numpy[0]
+
+perf = predictor.evaluate_predictions(y_test[:num_test], preds, auxiliary_metrics=True)
+print("Predictions: ", preds)
+
+predictor.unpersist_models() # free memory by clearing models, future predict() calls will load models from disk
+```
+
+You can alternatively specify a particular model to persist via the `models` argument of `persist_models()`, or simply set `models='all'` to simultaneously load every single model that was trained during `fit`.
+
+### Collapsing bagged ensembles via refit_full
+
+For a ensemble predictor trained with bagging (as done above), recall there ~10 bagged copies of each individual model trained on different train/validation folds. We can collapse this bag of ~10 models into a single model that's fit to the full dataset, which can greatly reduce its memory/latency requirements (but may also reduce accuracy). Below we refit such a model for each original model but you can alternatively do this for just a particular model by specifying the `model` argument of `refit_full()`.
+
+```{.python .input}
+refit_model_map = predictor.refit_full()
+print("Name of each refit-full model corresponding to a previous bagged ensemble:")
+print(refit_model_map)
+predictor.leaderboard(test_data, silent=True)
+```
+This adds the refit-full models to the leaderboard and we can opt to use any of them for prediction just like any other model. Note `pred_time_test` and `pred_time_val` list the time taken to produce predictions with each model (in seconds) on the test/validation data. Since the refit-full models were trained using all of the data, there is no internal validation score (`score_val`) available for them. You can also call `refit_full()` with non-bagged models to refit the same models to your full dataset (there won't be memory/latency gains in this case but test accuracy may improve).
+
+### Model distillation
+
+While computationally-favorable, single individual models will usually have lower accuracy than weighted/stacked/bagged ensembles. [Model Distillation](https://arxiv.org/abs/2006.14284) offers one way to retain the computational benefits of a single model, while enjoying some of the accuracy-boost that comes with ensembling. The idea is to train the individual model (which we can call the student) to mimic the predictions of the full stack ensemble (the teacher). Like `refit_full()`, the `distill()` function will produce additional models we can opt to use for prediction.
+
+```{.python .input}
+student_models = predictor.distill(time_limits=30) # specify much longer time-limits in real applications
+print(student_models)
+preds_student = predictor.predict(test_data_nolabel, model=student_models[0])
+print(f"predictions from {student_models[0]}:", preds_student)
+predictor.leaderboard(test_data)
+```
+
+### Faster presets or hyperparameters
```

Comment:
Consider adding another section talking about selecting models along the pareto frontier via 

```
predictor.fit_weighted_ensemblee(expand_pareto_frontier=True)
predictor.leaderboard(only_pareto_frontier=True)
```

Where leaderboard will contain the most accurate model for a given latency. Then the user can select a model with acceptable latency from the list.

Reply:
added new section "Using smaller ensemble or faster model for prediction"